WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt

```diff
@@ -58,7 +62,10 @@ class ReaderPostDetailViewModel @Inject constructor(
             currentUiState?.let {
                 findPost(currentUiState.postId, currentUiState.blogId)?.let { post ->
                     post.isFollowedByCurrentUser = data.following
-                    _uiState.value = currentUiState.copy(headerUiState = createPostDetailsHeaderUiState(post))
+                    _uiState.value = currentUiState.copy(
+                            moreMenuItems = null,
```

Comment:
Could you please elaborate why we need this line? It feels like a workaround but I might be missing something. Is it related to the popup window dismissal?

Reply:
Yes it is related to the dismissal. Menu should’ve been dismissed on clicking “follow” from it but it didn’t and got redisplayed as [menu items were not null](https://github.com/wordpress-mobile/WordPress-Android/blob/fa79e5581e43cfde3f4d801ecdd541c62619b530/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt#L425-L426) and I agree it looks like a workaround. I haven't found a better way to handle it yet. Let me know if you have suggestions.

Reply:
Could you see if this looks better in PR #13129?
https://github.com/wordpress-mobile/WordPress-Android/blob/354dff4942ebd18cbad6ec22cfd50773468f0c5b/WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt#L65

Reply:
👍 It's definitely better, thank you!

I think ideally, the follow would just copy the currentUiState and update what's necessary = follow state. However, I think this solution is good as well.